### ISSUE TITLE
MissingSwitchDefault: false positive with exhaustive switch expressions on enums (Java 14 + Java 15)

### ISSUE THREAD

**Issue by vovkss** [checkstyle/checkstyle#8809]
Check documentation: https://checkstyle.org/config_coding.html#MissingSwitchDefault

```
/var/tmp $ javac Switcher.java


/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MissingSwitchDefault"/>
  </module>
</module>


/var/tmp $ cat Switcher.java
public class Switcher {

    public enum Fruit {
        APPLE,
        PEAR
    }

    public int theSwitcher(final Fruit fruit) {
        return switch (fruit) {
            case APPLE -> 1;
            case PEAR -> 2;
        };
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.36-all.jar -c config.xml Switcher.java
Starting audit...
[ERROR] C:\Users\vovks\Desktop\xx\Switcher.java:9:16: switch without "default" clause. [MissingSwitchDefault]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

**Expecting no error,** since the cases of the `switch` expression are exhaustive. 

In fact, including a `default` clause would actually **harm** in this case: the compile wouldn't emit a compilation error if a new enum constant is introduced. 

It would possibly make sense to include a new separate check which ensures that the `default` clause is **not** included in exhaustive enum-based `switch` expressions.

**Comment by romani** [comment]
duplicate of https://github.com/checkstyle/checkstyle/issues/8779
just recently merged.
